lib/browser/desktop-capturer.js

```diff
@@ -28,45 +28,43 @@ ipcMain.on('ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', function (event, cap
 
   // If the WebContents is destroyed before receiving result, just remove the
   // reference from requestsQueue to make the module not send the result to it.
-  event.sender.once('destroyed', function () {
+  event.sender.once('destroyed', () => {
     request.webContents = null
+    return
   })
 })
 
-desktopCapturer.emit = function (event, name, sources) {
+desktopCapturer.emit = (event, name, sources) => {
   // Receiving sources result from main process, now send them back to renderer.
-  var handledRequest, i, len, ref, ref1, request, result, source, unhandledRequestsQueue
-  handledRequest = requestsQueue.shift(0)
-  result = (function () {
-    var i, len, results
-    results = []
-    for (i = 0, len = sources.length; i < len; i++) {
-      source = sources[i]
-      results.push({
-        id: source.id,
-        name: source.name,
-        thumbnail: source.thumbnail.toDataURL()
-      })
+  const handledRequest = requestsQueue.shift()
+  const handledWebContents = handledRequest.webContents
+  const unhandledRequestsQueue = []
+
+  const result = sources.map(source => {
+    return {
+      id: source.id,
+      name: source.name,
+      thumbnail: source.thumbnail.toDataURL()
     }
-    return results
-  })()
-  if ((ref = handledRequest.webContents) != null) {
-    ref.send('ELECTRON_RENDERER_DESKTOP_CAPTURER_RESULT_' + handledRequest.id, result)
+  })
+
+  if (handledWebContents != null) {
+    handledWebContents.send(capturerResult(handledRequest.id), result)
   }
 
   // Check the queue to see whether there is other same request. If has, handle
   // it for reducing redunplicated `desktopCaptuer.startHandling` calls.
```

Comment:
_(trivial)_ while you're in this part of the code, might as well fix this comment